staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -138,6 +148,48 @@ func (w *watcher) createWatchChan(ctx context.Context, key string, rev int64, re
 	return wc
 }
 
+// isInitialEventsEndBookmarkRequired since there is no way to directly set
+// opts.ProgressNotify from the API and the etcd3 impl doesn't support
+// notification for external clients we simply return initialEventsEndBookmarkRequired
+// to only send the bookmark event after the initial list call.
+//
+// see: https://github.com/kubernetes/kubernetes/issues/120348
+func (w *watcher) isInitialEventsEndBookmarkRequired(opts storage.ListOptions) (bool, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.WatchList) {
+		return false, nil
+	}
+	if opts.SendInitialEvents == nil || !*opts.SendInitialEvents || !opts.Predicate.AllowWatchBookmarks {
```

Comment:
nit:
```
return opts.SendInitialEvents != nil && *opts.SentInitialEvents && !opts.Predicate.AllowWatchBookmarks
```

Reply:
this seems to be incorrect. I think it should be

```
return opts.SendInitialEvents != nil && *opts.SendInitialEvents && opts.Predicate.AllowWatchBookmarks
```

Reply:
yeah sorry - that's what I meant